sklearn/neighbors/tests/test_ball_tree.py

```diff
@@ -25,6 +38,11 @@
     "sokalsneath",
 ]
 
+BALL_TREE_CLASSES = [
+    BallTree64,
+    BallTree32,
```

Comment:
I think it makes sense to test the public API.
```suggestion
    BallTree32,
    BallTree,
```

Reply:
Is this public api the one that is present inside `sklearn.neighbors.__init__`?

Reply:
Generally, the public API is the one documented and the one discoverable when using a REPL.

Programmatically, `__all__` in `__init__.py` files contains elements of the public API. 

Here `BallTree` is an element of the public API, but `BallTree32` and `BallTree64` aren't.

Reply:
Actually, my comments in this thread do not make sense since `BallTree` is a local alias of `BallTree32`. 

Those comments would only make sense in something proposed in is accepted.